### PR TITLE
Removing additional query parameters in SSO URL while signing

### PR DESCRIPTION
**Pull request by SruthiD-Invision**
KPMG is the first team that has a query parameter in their SSO URL: 
`https://intext.us.kpmg.com/FIM/sps/idpintext/saml20/login?Fed=invisionapp` and the signature validation is failing. 

We suspect that the failure is due to the query parameter being included in the signature. 
Fixing this to test the case. This doesn't affect other teams (without a query parameter in their SSO URL)

### CODE REVIEW

File: service_provider.go [hunk 1]
```diff
@@ -261,24 +261,30 @@ func (req *AuthnRequest) Redirect(relayState string, sp *ServiceProvider) (*url.
...
 	// We can't depend on Query().set() as order matters for signing
 	query := rv.RawQuery
+
 	if len(query) > 0 {
 		query += "&SAMLRequest=" + url.QueryEscape(string(w.Bytes()))
 	} else {
 		query += "SAMLRequest=" + url.QueryEscape(string(w.Bytes()))
 	}
 
+	queryToBeSigned := "SAMLRequest=" + url.QueryEscape(string(w.Bytes()))
```
Comment (CPWeaver) [inline service_provider.go:271]: could you not have simply removed lines 265-267? and is there a reason to preserve the `query` variable which is no longer used?

**Review by CPWeaver — APPROVED** [review]
This is ok as an experiment but if it is permanent I think it could use some cleanup.

File: service_provider.go (hunk 1 above)
Comment (SruthiD-Invision) [reply service_provider.go:271]: > could you not have simply removed lines 265-267? and is there a reason to preserve the `query` variable which is no longer used?

We are using query variable to send the original request line#291 and 294

Comment (CPWeaver) [reply service_provider.go:271]: ah, I see it right below the diff window. Thanks for clarifying!

**Pull request merged**